Update docs with getting started guide (#880)
FuelLabs/sway: docs/src/getting-started/deploy-and-call-with-cli.md

# Deploy and call a Sway contract with the fuel-core CLI

> ## 6. Spin up a fuel node

Reviewer: I suspect we should be using `Fuel` everywhere instead of `fuel`. Except, of course, in things like `fuel-core`.

Reviewer 2: ☝️

In order to deploy the contract, let's spin up a local fuel node:

```console
fuel-core --db-type in-memory
```

## 7. Deploy `wallet_contract` on your local fuel node

To deploy `wallet_contract` on your local fuel node, run

```console
forc deploy
```

from the root of the `wallet_contract` directory.

This should produce some output in stdout that looks like this:

```console

>   Compiled library "lib-std" with 7 warnings.

Reviewer 2: We can't really get rid of them until the compiler can handle things like Result and impl self and constants.

script;

use wallet_abi::Wallet;
use wallet_abi::SendFundsRequest;
use std::constants::ETH_ID;

fn main() {
    let caller = abi(Wallet, contract_address);
    let req = SendFundsRequest {
        amount_to_send: 200,
        recipient_address: 0x9299da6c73e6dc03eeabcce242bb347de3f5f56cd1c70926d76526d7ed199b8b,
    };
    caller.send_funds(10000, 0, ETH_ID, req);
}
````

Replace the `$contract_address` with the contract id you noted in step 8.

The main new concept is the _abi cast_: `abi(AbiName, contract_address)`. This returns a `ContractCaller` type which can be used to call contracts. The methods of the ABI become the methods available on this contract caller: `send_funds` and `receive_funds`. We then construct the request format, `SendFundsRequest`, and directly call the contract ABI method as if it was just a regular method.

## 9. Build `wallet_script`

To build `wallet_script`, run

```console
forc build
```

from the root of the `wallet_script` directory.

## 10. Run the `wallet_script` against the local fuel node

To run the script now against the local fuel node, run

```console
forc run --contract <contract-id>
```

from the root of the `wallet_script` directory.

> Note that we are passing in the `wallet_script` contract-id as a flag. You will need to pass in the contract id of every contract that this script will be interacting with.

Reviewer: ```suggestion
Note that we are passing in the contract id of `wallet_contract` using the flag `--contract`. You will need to pass in the contract id of every contract that this script will be interacting with.
```
